File: 04.Simulation模拟/leetcode_1189_maxNumberOfBalloons.py
from collections import Counter

class Solution:
    def maxNumberOfBalloons(self, text: str) -> int:
        origin = Counter("balloon")
        cnt = Counter(text)
        res = 10**18
        for i in origin.keys():
            res = min(res, cnt[i]//origin[i])
        return res

if __name__ == "__main__":
    s = Solution()
    text = "nlaebolko"
    print(s.maxNumberOfBalloons(text))

File: 04.Simulation模拟/leetcode_1189_maxNumberOfBalloons.cpp
#include <iostream>
#include <vector>
#include <unordered_map>
using namespace std;

class Solution {
public:
    int maxNumberOfBalloons(string text) {
        unordered_map<char, int> origin = {{'b',1},{'a',1},{'l',2},{'o',2},{'n',1}};
        unordered_map<char, int> cnt;
        for (auto& c : text){
            cnt[c]++;
        }
        int res = 1e9;
        for (auto& kv : origin){
            res = min(res, cnt[kv.first]/kv.second);
        }
        return res;
    }
};

int main(){
    Solution s;
    string text = "nlaebolko";
    cout << s.maxNumberOfBalloons(text) << endl;
}
